go/vt/vtctl/schematools/schematools.go

```diff
@@ -43,3 +47,66 @@ func GetSchema(ctx context.Context, ts *topo.Server, tmc tmclient.TabletManagerC
 
 	return sd, nil
 }
+
+// ParseSchemaMigrationStrategy parses the given strategy into the underlying enum type.
+func ParseSchemaMigrationStrategy(name string) (vtctldatapb.SchemaMigration_Strategy, error) {
+	if name == "" {
+		// backward compatiblity and to handle unspecified values
+		return vtctldatapb.SchemaMigration_DIRECT, nil
+
+	}
+
+	upperName := strings.ToUpper(name)
+	switch upperName {
+	case "GH-OST", "PT-OSC":
+		// more backward compatibility since the protobuf message names don't
+		// have the dash.
+		upperName = strings.ReplaceAll(upperName, "-", "")
+	default:
+	}
+
+	strategy, ok := vtctldatapb.SchemaMigration_Strategy_value[upperName]
+	if !ok {
+		return 0, fmt.Errorf("unknown schema migration strategy: '%v'", name)
+	}
+
+	if upperName != name {
+		// TODO (andrew): Remove special handling for lower/uppercase and
+		// gh-ost=>ghost/pt-osc=>ptosc support. (file issue for this).
+		log.Warningf("detected legacy strategy name syntax; parsed %q as %q. this will break in the next version.", upperName, name)
+	}
+
+	return vtctldatapb.SchemaMigration_Strategy(strategy), nil
+
+}
+
+// ParseSchemaMigrationStatus parses the given status into the underlying enum type.
+func ParseSchemaMigrationStatus(name string) (vtctldatapb.SchemaMigration_Status, error) {
+	key := strings.ToUpper(name)
+
+	val, ok := vtctldatapb.SchemaMigration_Status_value[key]
+	if !ok {
+		return 0, fmt.Errorf("unknown enum name for SchemaMigration_Status: %s", name)
+	}
+
+	if name != key {
+		log.Warningf("legacy status name %s parsed as %s. this will break in a future release.", name, key)
```

Comment:
Curious why we use %q above and %s here?

Reply:
i think i meant `%q` in both cases